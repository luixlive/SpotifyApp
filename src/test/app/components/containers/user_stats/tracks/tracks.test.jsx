import _ from 'lodash';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';

import {
  tracks as componentProps,
} from './../../../../../test_utils/components_props';
import initialState from './../../../../../test_utils/initial_state';
import Tracks, {
  PureTracks,
} from './../../../../../../app/components/containers/user_stats/tracks/tracks';

describe('App Components - Tracks', () => {
  const mockStore = configureStore();
  let store;
  beforeAll(() => {
    store = mockStore(initialState);
  });

  describe('Snapshots', () => {
    let props;
    beforeEach(() => {
      props = _.cloneDeep(componentProps);
    });

    it('renders no mobile', () => {
      const rendered = renderer.create((
        <Provider store={store}>
          <PureTracks {...props} />
        </Provider>
      )).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders mobile', () => {
      props.deviceMobile = true;
      const rendered = renderer.create((
        <Provider store={store}>
          <PureTracks {...props} />
        </Provider>
      )).toJSON();
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('Provider', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = mount((
        <Provider store={store}>
          <Tracks />
        </Provider>
      ));
    });

    it('renders', () => {
      expect(wrapper.find(PureTracks).length).toEqual(1);
    });

    it('matches initial state', () => {
      expect(Object.keys(wrapper.find(PureTracks).props()))
        .toEqual(Object.keys(componentProps));
    });
  });
});
