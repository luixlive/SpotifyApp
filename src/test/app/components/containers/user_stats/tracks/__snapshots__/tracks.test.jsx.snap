// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components - Tracks Snapshots renders 1`] = `
<div
  className="ui basic segment"
>
  <div
    className="ui grid"
  />
</div>
`;
